modules/codegen-plugin/src/smithy4s/codegen/Smithy4sCodegenPlugin.scala

```diff
@@ -403,28 +403,30 @@ object Smithy4sCodegenPlugin extends AutoPlugin {
       (inputDirs ++ generatedFiles)
         .filter(_.exists())
         .toList
-    val outputPath = (conf / smithy4sOutputDir).value
-    val resourceOutputPath = (conf / smithy4sResourceDir).value
+    val outputPath = (conf / smithy4sOutputDir).value / "smithy4s"
+    val resourceOutputPath = (conf / smithy4sResourceDir).value / "smithy4s"
```

Comment:
the changes to the target directories is the reason why the scripted SBT tests are failing. You're gonna have to amend a bunch of files called `test` in the direct subfolders of https://github.com/disneystreaming/smithy4s/tree/series/0.18/modules/codegen-plugin/src/sbt-test/codegen-plugin

Reply:
That's what I'm on right now. It takes a while to run those + I can't make `multimodule` case run yet, but I'll get there :crossed_fingers:

Reply:
you can run them individually if it helps

Reply:
I went through the scripted tests and adjusted the paths. Here's what seems to work and what not
```
✅ aws-newtype-flatten
✅ aws-open-enums
✅ aws-specs
✅ custom-settings
✅ defaults
✅ dependencies-only
✅ extra-configs
❌ multimodule
✅ multimodule-aws
✅ multimodule-no-compile
❌ multimodule-staged
✅ optics-generate
✅ protobuf
✅ scala3
✅ smithy-rules
❌ wildcard-config
```
This made me question the part about getting rid of hashing generated sources contents. If we have a look at the last test for example, here's what it does:
```
> ++2.13.10 root/compile
$ exists target/scala-2.13/src_managed/main/smithy/generated-metadata.smithy
$ exists target/scala-2.13/resource_managed/main/smithy4s/META-INF/smithy/generated-metadata.smithy

> ++3.3.0 root/compile
$ exists target/scala-3.3.0/src_managed/main/smithy/generated-metadata.smithy
$ exists target/scala-3.3.0/resource_managed/main/smithy4s/META-INF/smithy/generated-metadata.smithy

# ensure metadata file is re-generated after deleting
$ delete target/scala-2.13/src_managed/main/smithy/generated-metadata.smithy
$ delete target/scala-3.3.0/resource_managed/main/smithy4s/META-INF/smithy/generated-metadata.smithy
> ++3.3.0 root/compile
$ exists target/scala-3.3.0/src_managed/main/smithy/generated-metadata.smithy
$ exists target/scala-3.3.0/resource_managed/main/smithy4s/META-INF/smithy/generated-metadata.smithy
```
Why it does `delete target/scala-2.13/src_managed/main/smithy/generated-metadata.smithy` is unclear to me (i'd expect 3.3.0 there) but since my version doesn't do hashing and only remembers the path, it would not re-generate the sources in case they were modified by hand.

I think we need to decide if this fine or not. Without hashing managed sources and resources won't be regenerated which I think users will find confusing. Should I restore hashing and only stick with having the dedicated smithy4s managed directories?

Reply:
I think we can't just change the base path of `resources` because that messes with SPI and smithy's manifest (it has to be `META-INF/smithy/manifest`), that's why `multimodule` is failing (downstreams don't see shapes from upstreams)

Reply:
Changing this (removing the `smithy4s` nesting in resources) makes most testspass, but the one you've posted is still problematic.

IMO manually touching target files means you're on your own. Instead of this `delete`, we could do a `clean` or just skip that check entirely.

Reply:
Logically I agree, but I wonder why the test was added in the first place ... Let's exhume whatever PR added it, see if we can get a hint

Reply:
I'm looking into it.

Reply:
The test was added in https://github.com/disneystreaming/smithy4s/pull/736/.

I think it was just added as what seemed intuitive at the time. Removed these lines in 70709ae596a3613d6ad3aeac4445de70ea820ea2